### PR TITLE
Implement the Jinja logic for the desktop navigations dropdowns

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Implements the logic for rendering the 3 different dropdown views, [see figma](https://www.figma.com/design/p0y7lHJ7edLIeSXM1qbpRZ/23.10-C.com-%26-U.com---Navigation?node-id=455-9266&t=0kWDK2TbW11Y4Z0N-0)
- Makes some changes to the navigaiton.yaml to make processing it easier
- This led to some refactoring of how the mobile view is rendered
- Some refactoring to reuse some templates and move sections into their own templates
- move the SCSS into a separate file with separate mixins. This is mostly to make up-streaming it easier

## QA
**please ignore keyboard navigation, careers roles not appearing and the giant black boxes which will later be images, they are handled in separate tickets**

- Compare what I have to [the figma design](https://www.figma.com/design/p0y7lHJ7edLIeSXM1qbpRZ/23.10-C.com-%26-U.com---Navigation?node-id=455-9266&t=0kWDK2TbW11Y4Z0N-0)
- Make sure all the content is there and the links work etc
- Check the Jinja logic and Scss makes sense etc, check for repeating code or when things could be made clearer (naming, structure etc)

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-12670

## Screenshots

View 1:
image

View 2:
image

View 3:
image

### CODE REVIEW

File: static/js/modals.js [hunk 1]
```diff
@@ -219,7 +219,7 @@ function getCustomFields(event) {
       document.getElementById("advice")?.value
```
Comment (petesfrench) [inline static/js/modals.js:219]: This was from running 'format-prettier' - I didn't touch it

File: static/js/prepare-form-inputs.js [hunk 2]
```diff
@@ -34,7 +34,6 @@ function preFormatCountry(countryCode, countryInput) {
   countryInput.value = countryCode;
```
Comment (petesfrench) [inline static/js/prepare-form-inputs.js:34]: This was from running 'format-prettier' - I didn't touch it

File: static/sass/_pattern_homepage-modal.scss [hunk 3]
```diff
@@ -1,6 +1,7 @@
 .is-dark-modal {
-    .p-modal__dialog, .p-modal__header {
-        background-color: $colors--dark-theme--background-default;
-        color: $colors--dark-theme--text-default;
-    }
-}
\ No newline at end of file
+  .p-modal__dialog,
```
Comment (petesfrench) [inline static/sass/_pattern_homepage-modal.scss:2]: This was from running 'format-prettier' - I didn't touch it

File: templates/navigation/partials/_preview-links.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<div class="p-navigation__preview-links">
+  {% if preview_secondary_links.title %}
+    <p class="p-navigation--list-heading">{{ preview_secondary_links.title }}</p>
+  {% endif %}
+  <div class="row u-no-padding">
+    {% for link in preview_secondary_links.links %}
+      <div class="col-2">
+        <a class="p-link--inverted" href="">
```
Comment (britneywwc) [inline templates/navigation/partials/_preview-links.html:8]: The href links aren't being set here for the preview links

Comment (britneywwc) [inline templates/navigation/partials/_preview-links.html:8]: The href links aren't being set here for the preview links

File: templates/navigation/_dropdown.html [hunk 5]
```diff
@@ -29,76 +29,58 @@
...
-                  <li class="p-navigation--list-heading">
-                    <h2 class="p-muted-heading is-muted" data-level="2">{{ links_section.title }}</h2>
+                {% if nav_section.secondary_links.title %}
+                <li class="p-navigation--list-heading">
+                  <h2 class="p-muted-heading is-muted" data-level="2">{{ nav_section.secondary_links.title }}</h2>
+                </li>
+                {% endif %}
+                {% for link in nav_section.secondary_links.links %}
+                  <li>
+                    <a class="p-navigation--list-link" href="{{ link.url }}" data-level="2">{{ link.title }}</a>
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:47]: The href links are being added here for preview links, should it be removed or added to be consistent with `_preview_links.html`?

File: templates/navigation/_dropdown.html [hunk 6]
```diff
@@ -29,76 +29,58 @@
...
              style="margin-bottom:0"
              href="{{ section.cta_link }}"
              data-level="1">{{ section.cta }}</a>
         </li>
       {% endif %}
+
       {% if section.main_lists %}
-      <!-- Loop through each of the main lists -->
-        {% for main_list in section.main_lists %}
+        {% for list in section.main_lists %}
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:66]: Since `list.link_list` is being reused, can we set a variable for it? Something like 
`{% set linked_list = list.link_list %}`and use it in the lines below

File: templates/navigation/_dropdown.html [hunk 7]
```diff
@@ -29,76 +29,58 @@
...
           <li class="p-navigation--list-heading">
-            <h2 class="p-muted-heading" data-level="1">{{ main_list.title }}</h2>
+            <h2 class="p-muted-heading" data-level="1">{{ list.link_list.title }}</h2>
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:68]: After setting the variable
```suggestion
            <h2 class="p-muted-heading" data-level="1">{{ linked_list.title }}</h2>
```

File: templates/navigation/_dropdown.html [hunk 8]
```diff
@@ -29,76 +29,58 @@
...
           </li>
-          {% for link in main_list.link_list %}
-            <li>
-              <a class="p-navigation__dropdown-item" href="{{ link.url }}" data-level="1">{{ link.title }}</a>
-            </li>
+          {% for link in list.link_list.links %}
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:70]: Similarly here, to make it more explicit and readable
```suggestion
          {% for link in linked_list.links %}
```

**Review by britneywwc — COMMENTED** [review]
Left a few comments on missing href links for preview links and also setting variables. Looks great overall!

File: templates/navigation/partials/_preview-links.html (hunk 4 above)
Comment (petesfrench) [reply templates/navigation/partials/_preview-links.html:8]: Good catch 🙈

File: templates/navigation/_dropdown.html (hunk 5 above)
Comment (petesfrench) [reply templates/navigation/_dropdown.html:47]: Added, I just missed this

File: templates/navigation/_dropdown.html (hunk 6 above)
Comment (petesfrench) [reply templates/navigation/_dropdown.html:66]: This is a good point, its so confusing to read. I rewrote how the .yaml file structures these lists. I've opted to call them 'sibling lists' as I think this makes most since? anyway, take a look and tell me if it looks better

File: templates/navigation/partials/_preview-links.html [hunk 9]
```diff
@@ -5,9 +5,9 @@
   <div class="row u-no-padding">
     {% for link in preview_secondary_links.links %}
       <div class="col-2">
-        <a class="p-link--inverted" href="">
+        <a class="p-link--inverted" href="https://ubuntu.com/engage/sicredi-openstack-case-study">
           <div class="p-navigation--image-wrapper">
-            <img src="{{ link.image_url }}" alt="{{ link.alt_text }}" />
+            <!-- <img src="{% if link.image_url == '/PLACE-HOLDER' %}https://variety.com/wp-content/uploads/2021/07/Rick-Astley-Never-Gonna-Give-You-Up.png?w=1000&h=667&crop=1{% endif %}" alt="" /> -->
```
Comment (britneywwc) [inline templates/navigation/partials/_preview-links.html:10]: I think the link should be [this](https://www.youtube.com/watch?v=dQw4w9WgXcQ) instead

File: templates/navigation/_dropdown.html [hunk 10]
```diff
@@ -2,7 +2,7 @@
      data-level="1"
      id="{{ id }}-content"
      aria-hidden="true">
-  <div class="p-navigation__dropdown-content--sliding">
+  <div class="p-navigation__dropdown-content-mobile">
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:5]: Why was this changed from sliding to mobile? It's currently showing the mobile dropdown on desktop view

File: templates/navigation/_dropdown.html [hunk 11]
```diff
@@ -62,12 +62,12 @@ <h2 class="p-muted-heading is-muted" data-level="2">{{ nav_section.secondary_lin
         </li>
       {% endif %}
 
-      {% if section.main_lists %}
-        {% for list in section.main_lists %}
+      {% if section.sibling_lists %}
```
Comment (britneywwc) [inline templates/navigation/_dropdown.html:65]: Looks way cleaner! 🧊

File: templates/navigation/partials/_preview-links.html (hunk 9 above)
Comment (petesfrench) [reply templates/navigation/partials/_preview-links.html:10]: firstly, omg.. it's been a long time since I fell for that one.
secondly, oops forgot to remove my test images hahahaha

**Comment by petesfrench** [comment]
thanks @britneywwc, could you take another look pls

File: templates/navigation/_dropdown.html (hunk 10 above)
Comment (petesfrench) [reply templates/navigation/_dropdown.html]: There were some changes in vanilla I was trying to keep up with (as I am working off of main). I have reverted this now and it should be fixed

File: static/js/navigation/careers/Worker_populate-careers.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+/**
```
Comment (britneywwc) [inline static/js/navigation/careers/Worker_populate-careers.js:1]: Should the filename be all lowercase?

**Review by britneywwc — APPROVED** [review]
Thanks for making the changes, looks good to me! Just a minor question regarding the filename :)

File: static/js/navigation/careers/Worker_populate-careers.js (hunk 12 above)
Comment (petesfrench) [reply static/js/navigation/careers/Worker_populate-careers.js:1]: I wanted to emphasize that it a worker file. I haven't been able to find many conventions around naming workers in JS (I guess because it is a new technology). The one pattern I have seen is `populate-careers.worker.js`, but I also find trailing keywords often get missed.
